### ISSUE TITLE
[BUG/ISSUE] Ambiguity: planeflight_mod assumes inputs are Altitudes if "Type" input begins with the letter "S". 

### ISSUE THREAD

**Issue by jdhask** [geoschem/geos-chem#680]
This is not a bug and perhaps not worth addressing in the code, but its at least worth noting for other users. 

```planeflight_mod.F90``` decides to convert from altitude to pressure based on the input "Type" string passed with each point by preforming a string check against several NOAA Aircraft/Tower abbreviations listed on lines 991-1026. However, line 992 (and error check at 1605) checks not for an entire string match like every other line of the if statement checking for this condition, but it checks if the input "Type" string merely begins with an "S" which is supposed to indicate a surface site (` NAME(1:1) .EQ. 'S'`). If that condition is met, then it will assume you've passed altitudes and go on to adjust them to be pressures. 

```
! Convert from altitude to pressure if we have CCCG data or
       ! tower data
       NAME = ADJUSTL(PTYPE(N))
       IF ( NAME(1:1) .EQ. 'S'     .OR. & ! NOAA Surface                         *** Line 992 ***
            TRIM(NAME) .EQ. 'Aacg'   .OR. & ! NOAA Aircraft 
            TRIM(NAME) .EQ. 'Abne'   .OR. & ! NOAA Aircraft
            TRIM(NAME) .EQ. 'Acar'   .OR. & ! NOAA Aircraft
            TRIM(NAME) .EQ. 'Acma'   .OR. & ! NOAA Aircraft
            TRIM(NAME) .EQ. 'Acrv'   .OR. & ! NOAA Aircraft
...
			
          ! Change units
          ! NOTE: PHIS is now in units of [m], so we don't need to
          ! divide by g0 again (see issue geoschem/geos-chem #531)
          L_ALT = 0
          IJ = GET_IJ( LON, LAT, State_Grid )
          DO L = 1, State_Grid%NZ
             MOD_ELEV = State_Met%PHIS(IJ(1),IJ(2))                         &
                        + SUM( State_Met%BXHEIGHT(IJ(1),IJ(2),1:L) )
             IF ( (L_ALT .EQ. 0)  .AND. (MOD_ELEV .GT. PRES) ) THEN
                L_ALT = L
             ENDIF
          ENDDO
          PRES = State_Met%PMID(IJ(1),IJ(2),L_ALT)
       ENDIF
```

This may be an indented feature for different surface site inputs common for various NOAA sites, but the ambiguity comes in because this means any input "Type" string that begins with the letter "S" is assumed to have input from altitudes not pressures- which is not very intuitive. To be fair, it does print a warning to the log file & stop the run, if you've entered into this specific scenario and don't override the Stop call (though the error doesn't mention that its assuming your input to be altitudes...).  However, if you're running a simulation with the type string set to something that began with "S" using pressure inputs that could be at the surface occasionally thereby triggering this error, but not worrying users who expect to sample the surface  who allow the run to continue, then, the output PRESS columns can be different from the input PRESS columns since they were assumed to be altitudes and erroneously converted to pressures. 

***The best current work around is for users to simply avoid using input "Type" string names that begin with "S" if they are using pressure as inputs.*** A few possible fixes if others want could be to update line 992 to check for an entire string match (e.g. all surface sites are denoted as "S" only), to decide to convert from altitude inputs to pressure using the header line (use ALTS instead of PRES rather than "Type"), or even to update the warning (lines 1605-1611) to note that the vertical input is assumed to be an altitude because its a surface site & is being converted to a pressure.

**Comment by yantosca** [comment]
Thanks Jessica for writing.  The planeflight diagnostic is fairly old by this point and we don't include it in our benchmark testing.  That being said, it is still used by a lot of users, especially those who are looking at aircraft data (SEAC4RS, aTom, KORUS-AQ, etc).

I don't know if there is a good workaround that wouldn't clobber any prior behavior, so maybe your warning is the best workaround for users.  I'll leave this issue here so that it won't get automatically closed.

**Comment by msulprizio** [comment]
@jdhask The CH4 folks in our group added that a while back to account for NOAA surface observations, in which all station names begin with the letter S. When GEOS-Chem calls routine `PLANEFLIGHT` it should encounter this code and print a warning to the log file:

```
          ! If this is a surface observation, set L=1
          !
          ! NOAA Surface observations start with 'S' in Planeflight.dat
          ! Other surface observation strings can be added here
          NAME = ADJUSTL(PTYPE(M))
          IF ( NAME(1:1)  .EQ. 'S' ) THEN

             ! Make sure it is the users intention to set L=1
             IF ( Input_Opt%amIRoot ) THEN
                WRITE( 6, '(a)') 'WARNING: NOAA Surface Observation. '
                WRITE( 6, '(a)') 'Forcing L=1. If this is intended, '
                WRITE( 6, '(a)') 'you may comment out the call to '
                WRITE( 6, '(a)') 'GEOS_CHEM_STOP in routine '
                WRITE( 6, '(a)') 'PLANEFLIGHT (planeflight_mod.F90)'
                WRITE( 6, '(a)') REPEAT( '=', 79 )
             ENDIF
             CALL GEOS_CHEM_STOP

             ! Force L=1
             L = 1

          ENDIF
```

We can add a similar warning and stop in routine `READ_POINTS` to let users know that the code assumes you pass altitude and that will be converted to pressure. As with the other stop, if users are in fact passing surface data they can comment out the call to `GEOS_CHEM_STOP`, otherwise they can comment out the unit conversion. I can add this in 13.1.0.

**Comment by jdhask** [comment]
@msulprizio I think that would be a great easy fix- the current call to `GEOS_Chem_STOP` I actually overwrote when I was running older plane.log files with "Type"= 'SEAC4RS', 'SOAS', 'SENEX', etc  because I was expecting to occasionally sample the surface (e.g. if the plane was on the ground) & that's how I'd interpreted that warning.

**Comment by msulprizio** [comment]
This should now be fixed in https://github.com/geoschem/geos-chem/commit/334b22b910f908d3b0be4e01d53c622afa1432f0, which will be included in GEOS-Chem 13.1.0.
